src/main/scala/com/datastax/gatling/plugin/utils/CqlPreparedStatementUtil.scala

```diff
@@ -44,12 +45,80 @@ trait CqlPreparedStatementUtil {
   def getParamsList(preparedStatement: PreparedStatement): List[Int]
 }
 
+object SessionCollectionResolver {
+  def get(session:Session, name:String):Option[Any] = session.attributes.get(name)
+
+  def getClz[T <: Any](session:Session, name:String):Option[Class[T]] = {
+    get(session,name).flatMap((sessionVal) => {
+      sessionVal match {
+        case clz:Class[T] => Option(clz)
+        case _ => Option.empty
+      }
+    })
+  }
+
+  def getIterable[T <: Any](session:Session, name:String):Option[lang.Iterable[T]] = {
+    get(session,name).flatMap((sessionVal) => {
+      sessionVal match {
+        case rv:Iterable[T] => Option(rv.asJava)
+        case rv:lang.Iterable[T] => Option(rv)
+        case _ => Option.empty
+      }
+    })
+  }
+
+  def getMap[K <: Any, V <: Any](session:Session, name:String):Option[util.Map[K,V]] = {
+    get(session,name).flatMap((sessionVal) => {
+      sessionVal match {
+        case rv:Map[K,V] => Option(rv.asJava)
+        case rv:util.Map[K,V] => Option(rv)
+        case _ => Option.empty
```

Comment:
you can simplify API a lot by returning `Map.empty`, instead of `Option[Map[]]`
It is an anti-pattern to return the `Option` of Collection. The empty collection serves the purpose of Optional Already.
(Same for `getIterable` method)

Reply:
A variant of the discussion above, and I still disagree.  Option is warranted here because a value may not be present in the Gatling session.  If I were to try to use an empty Map to represent that I'd be unable to distinguish between a Session which had no mapping for a given property and a Session which mapped that property to an empty Map.

There's no harm in returning an Option containing a collection.  Option is just a monadic wrapper for a val... it can be a collection just as easily as it can be a scalar.

Reply:
When using `Optional` the  `Collections of Optionals` is considered an anti-patter and this has a good reason to be it. You don't need to distinguish between not present and empty collection here.

Reply:
It's perhaps a bit unusual but I've certainly seen cases where a collection of option types is very useful.  But regardless, that's not what we're doing here... we're returning an option of a collection.  As stated above the collection is just a value in this case; it's contents and/or it's type don't really impact the question of whether an option should be used.  It's true that a collection has a certain value (the "empty" collection) which can represent a specific state... but this _can_ be true for literally any type (if you use something like sentinel values).

As to your point about not needing to distinguish between not present and an empty collection... I don't think that's correct.  There are three distinct cases to consider here:

* Map is not present in the Gatling session
* Map is present in the Gatling session but is empty... therefore we can't get types off of it
* Map is present in the Gatling session and is not empty... so we can obtain types

Using the Option type to clearly represent these scenarios and throw precise exceptions which clearly indicate to the user what's going on.  If I follow your suggestion I can no longer do this.  That seems like a non-trivial price to pay for an API simplification.